discovery-infra/test_infra/controllers/node_controllers/node_controller.py

```diff
@@ -116,3 +116,27 @@ def set_ram_kib(self, node_name: str, ram_kib: int) -> None:
     @abstractmethod
     def get_machine_cidr(self) -> str:
         pass
+
+    @abstractmethod
+    def attach_interface(self, node_name, network_xml: str) -> Tuple[Any, Any]:
+        pass
+
+    @abstractmethod
+    def add_interface(self, node_name, network_name, target_interface: str) -> str:
+        pass
+
+    @abstractmethod
+    def undefine_interface(self, node_name: str, mac: str):
+        pass
+
+    @abstractmethod
+    def create_network(self, network_xml: str) -> str:
+        pass
+
+    @abstractmethod
+    def get_network_by_name(self, network_name: str) -> str:
+        pass
+
+    @abstractmethod
+    def destroy_network(self, network: Any):
```

Comment:
`Any`? What types do you expect over here?

Reply:
i expect to get libvirt network object

Reply:
So an integer or a string wouldn't work. Is there a class for that object? is that a dictionary?

Reply:
libvirt.virNetwork, ill import libvirt and use that

Reply:
fixed